Logger/BaseLoggerMixins.cs

```diff
@@ -2,6 +2,34 @@
 {
     public static class BaseLoggerMixins
     {
-        
+
+        public static void Warning(this BaseLogger logger, string message, params object[] args)
+        {
+            if (logger is null)
+                throw new System.ArgumentNullException();
```

Comment:
Typically when throwing an `ArgumentNullException` you want to specify the argument name.
```suggestion
                throw new System.ArgumentNullException(nameof(logger));
```